src/controller/java/src/chip/devicecontroller/OTAProviderDelegate.java

```diff
@@ -0,0 +1,121 @@
+/*
+ *   Copyright (c) 2023 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+package chip.devicecontroller;
+
+import javax.annotation.Nullable;
+
+public interface OTAProviderDelegate {
+  public QueryImageResponse handleQueryImage(
+      int vendorId,
+      int productId,
+      long softwareVersion,
+      @Nullable Integer hardwareVersion,
+      @Nullable String location,
+      @Nullable Boolean requestorCanConsent,
+      @Nullable byte[] metadataForProvider);
+
+  public void handleOTAQueryFailure(int error);
```

Comment:
please add doxygen for all java API, later we would generate these doxygen in github webpage..

Reply:
I added for doxygen.